src/py/mat3ra/made/tools/build/utils.py

```diff
@@ -0,0 +1,77 @@
+from typing import List
+
+import numpy as np
+from mat3ra.made.basis.basis import Basis
+from mat3ra.made.material import Material
+
+from mat3ra.made.tools.utils import convert_basis_to_crystal
+
+
+def resolve_close_coordinates_basis(basis: Basis, distance_tolerance: float = 0.01) -> Basis:
+    """
+    Find all atoms that are within distance tolerance and only keep the last one, remove other sites
+    """
+    coordinates = basis.coordinates.to_array_of_values_with_ids()
+    ids = set([coord.id for coord in coordinates])
+    ids_to_remove = set()
+
+    for i in range(1, len(coordinates)):
+        for j in range(i):
+            if np.linalg.norm(np.array(coordinates[i].value) - np.array(coordinates[j].value)) < distance_tolerance:
```

Comment:
We should have util functions that will allow to avoid using `np` many times

`get_distance_between_coordinates(coordinate1, coordinate2 = [0, 0, 0])`

`get_norm(vector) = get_distance_between_coordinates(vector, 0)`

similar to https://github.com/Exabyte-io/code/blob/65465971194255f85046161f97143d79658f7e89/src/js/math.ts#L15